core/executor/src/wasm_runtimes_pool.rs

```diff
@@ -0,0 +1,443 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implements a pool of pre-created runtime module instances.
+//!
+//! A background thread adds instances back to the pool and ensures
+//! a specified capacity is guaranteed.
```

Comment:
We cannot actually guarantee a specific capacity.

```suggestion
//! a specified capacity is maintained.
```